packages/expo-video/android/src/main/java/expo/modules/video/VideoManager.kt

```diff
@@ -19,11 +21,14 @@ object VideoManager {
   // Keeps track of all existing VideoPlayers, and whether they are attached to a VideoView
   private var videoPlayersToVideoViews = mutableMapOf<VideoPlayer, MutableList<VideoView>>()
 
+  private var playersRequestingKeepAwake = hashSetOf<HashableWeakReference<VideoPlayer>>()
```

Comment:
This construction can lead to some leaks. Here's a correct implementation of such a structure: https://gist.github.com/ZacSweers/d870015a3a86415a054b73816eada542 

When the weak reference is deallocated, it won’t be removed from the set. You might also be experiencing a similar issue with `fullscreenPlayerActivities` in `VideoManager`.

On another note, it feels like a bit of an overkill for what you’re trying to achieve 🤔